dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java

```diff
@@ -0,0 +1,113 @@
+
+package uk.ac.ebi.owl2json.xrefs;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import javax.annotation.RegEx;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class Bioregistry {
+
+    Gson gson = new Gson();
+
+    public String registryUrl;
+
+    JsonObject theRegistry;
+    Map<String, JsonObject> prefixToDatabase = new HashMap<>();
+    Map<String, Pattern> patterns = new HashMap<>();
+
+    public Bioregistry() {
+        this("https://raw.githubusercontent.com/biopragmatics/bioregistry/main/exports/registry/registry.json");
+    }
+
+    public Bioregistry(String jsonUrl){
+
+        this.registryUrl = jsonUrl;
+
+        try {
+            theRegistry = urlToJson(jsonUrl).getAsJsonObject();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        for(var entry : theRegistry.entrySet()) {
+            JsonObject db = entry.getValue().getAsJsonObject();
+            prefixToDatabase.put(db.get("preferred_prefix").getAsString(), db);
+        }
+    }
+
+    public String getRegistryUrl() {
+        return  registryUrl;
+    }
+
+    public String getUrlForId(String databaseId, String id) {
+
+        JsonObject db = prefixToDatabase.get(databaseId);
+
+        if(db == null)
+            return null;
+
+        if(id == null)
+            return null;
+
+        JsonElement patternObj = db.get("pattern");
+
+        if(patternObj == null)
+            return null;
+
+        Pattern pattern = getPattern(patternObj.getAsString());
+
+        if(!pattern.matcher(id).matches()) {
+            return null;
+        }
+
+        return db.get("uri_format").getAsString().replace("$1", id);
```

Comment:
should check that uri_format is not null - not all records in bioregistry have a uri_format

Reply:
Thanks, done!